### PR TITLE
guiceGame: implementing guice in a simple game.

### PR DESCRIPTION
**Pull request by VasilMitovClouway**
Using Guice dependency injection in various ways in a simple game. There are 3 layers of the game:
- The top layer is the Duel object which depends on having two participants.

- The participants are Charecter objects who depend on having one of two sets of Armor implementations and a random skill from the SkillProvider which is bound to a list of skills.

- The third layer are the Skill and Armor interfaces. Multibinding is implemented in there.
There are interceptors for Fireball cast() method and in sword attack. 
The Fireball object is an EagerSingleton and just to test things out I bound the Duel object to a normal/lazy singleton.

### CODE REVIEW

File: src/main/java/com/clouway/guice/modules/SkillModule.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+import java.util.Random;
+
+/**
+ * @author Vasil Mitov <v.mitov.clouway@gmail.com>
+ */
+public class SkillModule extends AbstractModule {
+
+  protected void configure() {
+    MapBinder<Integer, Skill> mapBinder = MapBinder.newMapBinder(binder(), Integer.class, Skill.class);
```
Comment (mgenov) [inline src/main/java/com/clouway/guice/modules/SkillModule.java]: Doesn't look like a correct example for MapBinder as binding to simple map to instances would do the same job.

Comment (VasilMitovClouway) [inline src/main/java/com/clouway/guice/modules/SkillModule.java]: Changes to the VillianCharecter. Now it has a Set of Skills that a SkillSetProvider provides from a SkillSetModule.

Comment (mgenov) [inline src/main/java/com/clouway/guice/modules/SkillModule.java]: Still MapBinder is not used correctly.

File: src/main/java/com/clouway/guice/modules/SkillSetModule.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import com.google.inject.multibindings.Multibinder;
+
+import java.util.Set;
+
+/**
+ * @author Vasil Mitov <v.mitov.clouway@gmail.com>
+ */
+public class SkillSetModule extends AbstractModule {
+  protected void configure() {
+    Multibinder<Skill> multibinder = Multibinder.newSetBinder(binder(),Skill.class);
```
Comment (mgenov) [inline src/main/java/com/clouway/guice/modules/SkillSetModule.java]: The problem is same as with Mapbinder. Doesn't really show a good usage.

File: src/main/java/com/clouway/guice/modules/SkillModule.java (hunk 1 above)
Comment (mgenov) [inline src/main/java/com/clouway/guice/modules/SkillModule.java]: 1. Try to bind <code>Map<Integer, Skill></code> in ModuleA and MobuleB then try: Guice.createInjector(new ModuleA(), new ModuleB()); 
2. Try the same with multibinder

Comment (VasilMitovClouway) [inline src/main/java/com/clouway/guice/modules/SkillModule.java]: Now there are two modules bound with sets for skills. The Villian takes the skills from both modules. Now I get the idea of binding multiple implementations. If it were with map it wouldnt work.

**Comment by VasilMitovClouway** [comment]
PR reminder.

**Pull request merged**